### PR TITLE
Fix alignment detection for long integers on 32-bits platforms

### PR DESCRIPTION
**Pull request by shindere**
This should fix Issue #7938.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -698,14 +698,13 @@ AC_CHECK_ALIGNOF([double])
...
-    [AS_IF([test "$target_cpu" != "i686" ],
-      [AC_DEFINE([ARCH_ALIGN_DOUBLE], [1])])])
-  AS_IF([test x"$ac_cv_alignof_long" = "x8" ],
-    [AC_DEFINE([ARCH_ALIGN_INT64], [1])]
-  )
-  ]
-)
+  [AS_CASE([$target_cpu],
+    [i686], [],
+    [AS_IF([test x"$ac_cv_alignof_double" = "x8" ],
```
Comment (xavierleroy) [inline configure.ac]: Technically, you want to define `ARCH_ALIGN_DOUBLE` if the alignment of type `double` is greater than one word, namely 4 in the case of a 32-bit platform.  Imagine a weird platform where `double` must be 16-aligned...  To deal with this unlikely case, you may want to write
```
test "$ac_cv_alignof_double" -gt 4
```

File: configure.ac [hunk 2]
```diff
@@ -698,14 +698,13 @@ AC_CHECK_ALIGNOF([double])
...
+      [AC_DEFINE([ARCH_ALIGN_DOUBLE], [1])])
+    AS_IF([test x"$ac_cv_alignof_long" = "x8" ],
```
Comment (xavierleroy) [inline configure.ac]: Likewise, you may want to write
```
test "$ac_cv_alignof_long" -gt 4
```

**Review by xavierleroy — APPROVED** [review]
Looks good to me.  One suggested clarification below, which should not make any difference on the usual architectures.  Feel free to use or not.

**Comment by shindere** [comment]
Many thanks for your review, @xavierleroy.

I implemented your suggestion because I liked it a lot and rebased to
resolve the conflict in Changes.

Before that CI was happy so merging.

**Comment by shindere** [comment]
Merged in trunk and cherry-picked to 4.08 as commit
6243c8787f5a8cab0b5bc6a1540ae5794de923a1

**Pull request merged**